# Chapter11 統計的仮説検定

- 公表されているフライドポテトの重さの平均 130g
- サンプルサイズ14個の標本平均 128.451g
- 14個の標本平均 128.451g は、ただの偶然か？

In [2]:
# ライブラリーの準備
import numpy as np
import pandas as pd
from scipy import stats

%precision 3
np.random.seed(1111)

In [3]:
# データの準備
df = pd.read_csv('../data/ch11_potato.csv')
sample = np.array(df['重さ'])
sample

array([122.02, 131.73, 130.6 , 131.82, 132.05, 126.12, 124.43, 132.89,
       122.79, 129.95, 126.14, 134.45, 127.64, 125.68])

In [4]:
# 標本平均の求める
s_mean = np.mean(sample)
s_mean

128.451

## 11.1 | 統計的仮説検定とは

- 統計的仮説検定（statistical hypothesis testing）
  - 母集団の母数に関する2つの仮設をたて、標本から計算される統計量を用いてどちらの仮設が正しいかを判断する統計的な枠組み。

### 11.1.1 統計的仮説検定の基本

- 母平均が 130g より少ないかどうか？
- フライドポテトの母集団が「正規分布」にしたがっていて、母分散が「9」とする。
- 仮説検定では、
  - 「母平均が130g」と仮定する。
  - この仮定のもとで、14個の標本は$X_1, X_2, ..., X_14 \sim^i.i.d. N(130, 9)$ にしたがう。
  - 標本平均 $\overline{X}$ は $N(130, 9/14)$ にしたがうことになる。
  - ここで、標本平均 $\overline{X}$ が $P(\overline{X} \leq x) = 0.05$ を満たす $x$ を考える。

In [5]:
rv = stats.norm(130, np.sqrt(9/14))
rv.isf(0.95)

128.681

- $P(\overline{X} \leq 128.681) = 0.05$ となる。
- 標本平均が「128.681g」以下の重さになることは「5%」の確率で発生する。
- Aさんが抽出した標本平均が「128.451g」となったのは、5%の確率で発生する出来事と言える。
- 仮説検定の大枠の考え方
  - 「標本平均が 128.451g という 128.681g 以下の重さとなった」のは、「母平均 130g」という仮説のもとで「5% の確率でしか発生しない出来事」が偶然起きたのではなく、もともと「母平均が 130g より少ない」と考える。
  - これによって、「母平均が 130g より少ない」と結論づける。

- 仮説検定では母数に関する2つの仮説をたてる。
  - 帰無仮説（null hypothesis）$H_0$
    - 対立仮説とは逆の仮説。「差がない」、「効果がない」
  - 対立仮説（alternative hypothesis）$H_1$
    - 主張したい仮説。「差がある」、「効果がある」
- 仮説検定から得られる結論
  - 帰無仮説を棄却する（reject the null hypothesis）
    - 「帰無仮説は正しくない」という結論。
  - 帰無仮説を採択する（accept the null hypothesis）
    - 「帰無仮説が正しくないとはいえない」という解釈
    - 「帰無仮説が正しいかどうかわからない」という保留の結論
  - 帰無仮説の「棄却・採択」の判断
    - 帰無仮説を仮定したもとで標本から計算される統計量が珍しい値がどうかで判断する。
    - 珍しい値が得られたら、「それは偶然ではなく何か意味のある値だ」と考えて帰無仮説を棄却する。
    - そうでなければ、帰無仮説を採択する。
  - 有意である（significant）
    - 偶然ではなく何か意味がるということ。
  - 棄却域（rejection region）
    - 帰無仮説が棄却される区間のこと
  - 採択域（acceptance region）
    - 帰無仮説が採択される区間のこと
  - 有意水準（level of significance）
    - 棄却域に入る確率
  - 臨界値（critical value）
    - しきい値
  - 検定統計量（test statistic）
    - 検定に使われる統計量
  - p値（p-value）
    - 検定統計量より左側の領域の面積
- フライドポテトの例では、
  - 有意水準: 5%
  - 検定統計量: 標本平均
  - 臨界値: 128.681

フライドポテトの重さの平均値について仮説検定を考える

- 帰無仮説「母平均は 130g」を仮定
- 14個のフライドポテトは互いに独立に $N(130, 9)$ にしたがう。
- 標本平均 $\overline{X}$ は $N(130, 9/14)$ にしたがう。
- 一般化して説明するため、標本平均 $\overline{X}$ を標準化する。
  - $Z = (\overline{X} - 130) / \sqrt{\frac{9}{14}}$
- 標準化することで上側$100\alpha\%$点を $z_alpha$ で表すことができる。
- 臨界値 $P((\overline{X} - 130)/\sqrt{\frac{9}{14}} \leq x) = 0.05$ を満たす $x$、すなわち $x = z_{0.95}$ を求めることができる。
- $(\overline{X} - 130)/\sqrt{\frac{9}{14}} \leq  z_{0.95}$ であれば、帰無仮説を棄却する。
- $(\overline{X} - 130)/\sqrt{\frac{9}{14}} \geq  z_{0.95}$ であれば、帰無仮説を採択する。

In [6]:
# 検定統計量を求める
z = (s_mean - 130) / np.sqrt(9/14)
z

-1.932

In [7]:
# 臨界値を求める
rv = stats.norm()
rv.isf(0.95)

-1.645

- 検定統計量と臨界値を比較する
  - 検定統計量のほうが小さい値になっている。
  - 帰無仮説が棄却される。
  - 平均は 130g より少ないという結論になる。

In [8]:
# p値を求める
rv.cdf(z)

0.027

- p値が 0.027 という有意水準 0.05 より小さい値になったので、帰無仮説は棄却される。

- 仮説検定の流れ
  - 仮説をたてる（帰無仮説・対立仮説）
  - 有意水準を決める
  - 検定統計量を計算する
  - p値を計算する
  - 有意水準より大きいか？
    - yes: 帰無仮説を採択
    - no: 帰無仮説を棄却